config/config.go

```diff
@@ -164,10 +179,13 @@ func (c *Config) parseConfig(ver protocolVersion) error {
 		Listener: listenAddr,
 		Plugins:  plugins,
 	}
+
 	if ver == protocolV6 {
 		c.Server6 = &sc
+		atomic.SwapPointer((*unsafe.Pointer)(unsafe.Pointer(&c.Server6)), unsafe.Pointer(c.Server6))
```

Comment:
What is actually being swapped here? Isn't the new config being assigned on the previous line?

Reply:
I also do not understand what is happening here, but I have tested it and it works properly.